### PR TITLE
Fix error messages compliance triggers

### PR DESCRIPTION
**Pull request by josepfo**

### CODE REVIEW

**Comment by josepfo** [comment]
[Correct validation messages in Compliance triggers](https://trello.com/c/fpECT4Lh/663-correct-validation-messages-in-compliance-triggers)

**Review by chaotixkilla — APPROVED** [review]

File: new-lamassu-admin/src/pages/Triggers/helper.js [hunk 1]
```diff
@@ -119,24 +108,66 @@ const Schema = Yup.object()
...
     }
-  )
+
+    return (
+      (triggerType && thresholdValidator?.[triggerType](threshold)) ||
+      context.createError({
+        path: 'threshold',
+        message: errorMessages?.[triggerType](threshold)
+      })
+    )
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Triggers/helper.js]: This is complicated to read. 

```js
if (triggerType && thresholdValidator[triggerType](threshold)) return

return context.createError({
  path: 'threshold',
  message: errorMessages[triggerType](threshold)
})
```

File: new-lamassu-admin/src/pages/Triggers/helper.js [hunk 2]
```diff
@@ -119,24 +108,66 @@ const Schema = Yup.object()
...
+        ? requirement.suspensionDays > 0
+        : true
+
+    return (
+      (requirement && requirementValidator(requirement)) ||
+      context.createError({
+        path: 'requirement',
+        message: 'Suspension days must be greater than 0'
+      })
+    )
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Triggers/helper.js]: same as above

File: new-lamassu-admin/src/pages/Triggers/helper.js [hunk 3]
```diff
@@ -119,24 +108,66 @@ const Schema = Yup.object()
...
+      txVelocity: threshold => {
+        const thresholdMessage = 'Transactions must be greater than 0'
+        const thresholdDaysMessage = 'Days must be greater than 0'
+        let message = ''
+        if (threshold.threshold <= 0) message = message.concat(thresholdMessage)
+        if (threshold.thresholdDays <= 0)
+          message = message
+            ? message.concat(', ' + thresholdDaysMessage)
+            : message.concat(thresholdDaysMessage)
+        console.log(message)
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Triggers/helper.js]: remove console.log

File: new-lamassu-admin/src/pages/Triggers/helper.js [hunk 4]
```diff
@@ -119,24 +108,66 @@ const Schema = Yup.object()
...
+      txVolume: threshold => {
+        const thresholdMessage = 'Volume must be greater than or equal to 0'
+        const thresholdDaysMessage = 'Days must be greater than 0'
+        let message = ''
+        if (threshold.threshold < 0) message = message.concat(thresholdMessage)
+        if (threshold.thresholdDays <= 0)
+          message = message
+            ? message.concat(', ' + thresholdDaysMessage)
+            : message.concat(thresholdDaysMessage)
+        console.log(message)
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Triggers/helper.js]: remove console.log

File: new-lamassu-admin/src/pages/Triggers/helper.js [hunk 5]
```diff
@@ -119,24 +108,66 @@ const Schema = Yup.object()
...
-        (triggerType && validator?.[triggerType](threshold)) ||
-        context.createError({ path: 'threshold' })
-      )
+  .test(({ threshold, triggerType }, context) => {
+    const errorMessages = {
+      txAmount: threshold => 'Amount must be greater than or equal to 0',
+      txVolume: threshold => {
+        const thresholdMessage = 'Volume must be greater than or equal to 0'
+        const thresholdDaysMessage = 'Days must be greater than 0'
+        let message = ''
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Triggers/helper.js]: Maybe it's better to have message be a array and then join the array with a , as separator. With this we can ditch the let for a const.

File: new-lamassu-admin/src/pages/Triggers/helper.js [hunk 6]
```diff
@@ -119,24 +108,66 @@ const Schema = Yup.object()
...
+        if (threshold.threshold < 0) message = message.concat(thresholdMessage)
+        if (threshold.thresholdDays <= 0)
+          message = message
+            ? message.concat(', ' + thresholdDaysMessage)
+            : message.concat(thresholdDaysMessage)
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Triggers/helper.js]: With the change above we can also remove this logic from here.

File: new-lamassu-admin/src/pages/Triggers/helper.js [hunk 7]
```diff
@@ -119,24 +108,66 @@ const Schema = Yup.object()
...
+      },
+      txVelocity: threshold => {
+        const thresholdMessage = 'Transactions must be greater than 0'
+        const thresholdDaysMessage = 'Days must be greater than 0'
+        let message = ''
+        if (threshold.threshold <= 0) message = message.concat(thresholdMessage)
+        if (threshold.thresholdDays <= 0)
+          message = message
+            ? message.concat(', ' + thresholdDaysMessage)
+            : message.concat(thresholdDaysMessage)
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Triggers/helper.js]: same as above

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

File: new-lamassu-admin/src/pages/Triggers/helper.js [hunk 8]
```diff
@@ -119,24 +108,59 @@ const Schema = Yup.object()
...
+    const errorMessages = {
+      txAmount: threshold => 'Amount must be greater than or equal to 0',
+      txVolume: threshold => {
+        const thresholdMessage = 'Volume must be greater than or equal to 0'
+        const thresholdDaysMessage = 'Days must be greater than 0'
+        const message = [thresholdMessage, thresholdDaysMessage]
+        if (threshold.threshold < 0 && threshold.thresholdDays <= 0)
+          return message.join(', ')
+        if (threshold.threshold < 0) return message[0]
+        if (threshold.thresholdDays <= 0) return message[1]
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Triggers/helper.js]:
```suggestion
        const message = []
        
        if (threshold.threshold < 0) message.push(threhsoldMessage)
        if (threshold.thresholdDays <= 0) message.push(thresholdDaysMessage)
        
        return message.join(', ')
```

File: new-lamassu-admin/src/pages/Triggers/helper.js [hunk 9]
```diff
@@ -119,24 +108,59 @@ const Schema = Yup.object()
...
+      },
+      txVelocity: threshold => {
+        const thresholdMessage = 'Transactions must be greater than 0'
+        const thresholdDaysMessage = 'Days must be greater than 0'
+        const message = [thresholdMessage, thresholdDaysMessage]
+        if (threshold.threshold <= 0 && threshold.thresholdDays <= 0)
+          return message.join(', ')
+        if (threshold.threshold <= 0) return message[0]
+        if (threshold.thresholdDays <= 0) return message[1]
+        return message
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Triggers/helper.js]: same as above

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**